### ISSUE TITLE
When moving a primary key, an unrunnable migration is generated

### ISSUE THREAD

**Issue by sevenseacat** [ash-project/ash_postgres#137]
**Describe the bug**

I just had the case where the primary key for a table was defined on the wrong attribute - on a FK named `experiment_id` instead of an attribute named `id`. So in my resource, I removed the primary key-ness from one field and added it to the other:

<img width="364" alt="Screenshot 2023-04-11 at 4 40 10 pm" src="https://user-images.githubusercontent.com/543859/231104962-05332e76-5f2d-4619-8c33-f15475426d68.png">

This generated a migration like the following:

```elixir
  def up do
    alter table(:experiment_instances) do
      add :id, :uuid, null: false, default: fragment("uuid_generate_v4()"), primary_key: true
      modify :experiment_id, :uuid
    end
  end

  def down do
    alter table(:experiment_instances) do
      modify :experiment_id, :uuid, primary_key: true
      remove :id
    end
  end
```

However this migration is not runnable - 

```
16:36:01.517 [info] alter table experiment_instances
** (Postgrex.Error) ERROR 42P16 (invalid_table_definition) multiple primary keys for table "experiment_instances" are not allowed
    (ecto_sql 3.9.2) lib/ecto/adapters/sql.ex:913: Ecto.Adapters.SQL.raise_sql_call_error/1
    (elixir 1.14.3) lib/enum.ex:1658: Enum."-map/2-lists^map/1-0-"/2
```

I can see a couple of issues with it - 

* It adds the new PK before removing the old one
* Temporarily removing the `id`-related lines,  the `modify` in `up` doesn't actually remove the existing PK index (meaning that the migration can't be rolled back, because the `down` will try to add a second PK which again is a no-no)

**Expected behavior**

The generated migration should remove the old PK index, then create a new one on the new field.

** Runtime
 - Elixir version 1.14.3
 - Erlang version 25.2
 - OS macOS Ventura
 - Ash version 2.6.27
 - any related extension versions
   - AshPostgres 1.3.18

**Comment by zachdaniel** [comment]
Well, I thought this would be a quick fix, but it can't ever be that simple can it 😆 

The problem is obvious in retrospect, but comes along with a fair amount of issues. Primarily, any foreign keys pointing at this column must also be dropped, then the primary key must be dropped, then the table must be altered with the new attributes set as primary key (I'm also having it leave in the old attribute but with `primary_key: false` for clarity on re-reading, even though technically its a noop since we already droped the pkey), and then recreate the references.

**Comment by zachdaniel** [comment]
This is fixed in main 525dcc9

**Comment by sevenseacat** [comment]
oh wow the fix for that is complex! Thank you for looking at it so quickly 🙌
